src/dispatch/plugins/dispatch_slack/views.py

```diff
@@ -223,8 +224,44 @@ def after_hours(user_email: str, incident_id: int, event: dict = None, db_sessio
             db_session.commit()
 
 
+def filter_tasks_by_assignee_and_creator(tasks: List[Task], by_assignee: str, by_creator: str):
+    """Filters a list of tasks looking for a given creator or assignee."""
+    filtered_tasks = []
+    for t in tasks:
+        if by_creator:
+            creator_email = t.creator.individual.email
+            if creator_email == by_creator:
+                filtered_tasks.append(t)
+
+        if by_assignee:
+            assignee_emails = [a.individual.email for a in t.assignees]
+            if by_assignee in assignee_emails:
+                filtered_tasks.append(t)
+
+    return filtered_tasks
```

Comment:
Should we move this to the task service and make it a db query instead?

Reply:
We could but the query will be fairly complex and the task lists are normally small so filtering after the fact seems reasonable.